### DISCUSSION TITLE
Unable to use workflow_dispatch for required status checks

### DISCUSSION THREAD

**Discussion by dahlbaek** [community/community#32913] [Actions] [answered]
We recently tried changing a workflow to running on workflow_dispatch events rather than on push events. One of the jobs in the given workflow is required to pass before merging to our main branch. It appears that status check remains in expected state, even though one has triggered a manual run of the workflow following https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow.

Is this behavior expected, or are we doing something wrong? Is there another way to put a required status check behind an approval button?

**Comment by airtower-luna** [comment, marked as answer]
That is expected, `workflow_dispatch` runs aren't associated with the PR. `push` and `pull_request` runs are.

You should be able to use an [environment with a protection rule](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) to require approval for a workflow to run e.g. on `push`, but it's not something I've tried myself.

**Reply by dahlbaek** [reply]
I see. We originally tried to use environments, but it seems the "Required Reviewers" protection rule requires you to either have a public repo or an Team/Enterprise subscription 🤔 I'll leave the question open for a while, to see if any other options pop up for putting required status checks behind approval buttons
